eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -325,7 +325,7 @@ object Graph {
         val totalWeight = prev.weight + (fee + hopCost).toLong * factor
         RichWeight(totalAmount, prev.length + 1, totalCltv, 1.0, totalFees, 0 msat, totalWeight)
       case Right(heuristicsConstants) =>
-        val hopCost = nodeFee(heuristicsConstants.hopCost, prev.amount)
+        val hopCost = if (edge.desc.a == sender) 0 msat else nodeFee(heuristicsConstants.hopCost, prev.amount)
```

Comment:
Would it make sense to apply the same `if` to the `failureCost` below?

Reply:
It's unnecessary. All paths will have the same `failureCost` for the first edge. Actually it's also unnecessary here for `hopCost`, I'll remove it.